Common Vulnerability from Plutonomicon and MLabs (#11)
input-output-hk/Certification-working-group: CommonVulnerabilities.md

[TODO]

> <ul>
> 	<li>Check if all are pertinent for a Vulnerability list</li>
> 	<li>Check if there is no double entry</li>
> 	<li>Check if some CPVs should be merged into a one</li>
> 	<li>Standardize the format of the entries</li>
> 	<li>Provide examples?</li>
> 	<li>Long term: Define a process for proposing a new CVE, reviewing it and adding it to the list</li>
> </ul>

Reviewer: Any particular reason why to use HTML here?

```suggestion

- Check if all are pertinent for a Vulnerability list
- Check if there is no double entry
- Check if some CPVs should be merged into a one
- Standardize the format of the entries
- Provide examples?
- Long term: Define a process for proposing a new CVE, reviewing it and adding it to the list
```

[References]

> <ul>
> 	<li>plutonimicon link</li>
> 	<li>MLabs audit</li>
> </ul>

Reviewer: Ditto.

```suggestion
- plutonimicon link
- MLabs audit
```

> | ID |CPV-2023-001  |
> |--|--|
> | Name | UTXO Value Size Spam AKA Token Dust Attack
> Description | The UTXO of too many tokens_  (or a single AssetClass with a large amount of tokens) - where a single utxo carries hundreds of unique tokens with different CurrencySymbols and/or TokenNames until it’s representation approaches the 16kb limit, this is then placed in a Validator in such a way that one or more Redeemers will need to consume this utxo, blocking transactions on that Redeemer/Validator
> |Mitigations| <ul><li>A minimum ada value is required on all UTXOs, which scales with UTXO size. This may mitigate protocol tx sizes somewhat. A similar calculation could be used on-chain to produce a size heuristic.</li><li>Don’t allow arbitrary value to be put in trusted places.</li></ul>|

Reviewer: This format makes things pretty hard to review. What about using several lines for lists of names or lists of mitigations? For instance (but doesn't have to be this):

```suggestion
| ID | CPV-2023-001 |
|--|--|
| Names       | UTXO Value Size Spam |
|             | Token Dust Attack |
| Description | The UTXO of too many tokens_  (or a single AssetClass with a large amount of tokens) - where a single utxo carries hundreds of unique tokens with different CurrencySymbols and/or TokenNames until it’s representation approaches the 16kb limit, this is then placed in a Validator in such a way that one or more Redeemers will need to consume this utxo, blocking transactions on that Redeemer/Validator |
| Mitigations | A minimum ada value is required on all UTXOs, which scales with UTXO size. This may mitigate protocol tx sizes somewhat. A similar calculation could be used on-chain to produce a size heuristic. |
|             | Don’t allow arbitrary value to be put in trusted places. |
```

I think the alignment of columns and the separation into multiple lines for the list makes for, in particular, much easier GitHub review. Such changes should probably be implemented prior to having everyone review this.

Author: Sorry for the formatting, I used an online markdown formatted that output this. I'll do something better.
